### ISSUE TITLE
Discussion: use a wrapper to provide multiple solc versions

### ISSUE THREAD

**Issue by gsalzer** [smartbugs/smartbugs#101]
Currently, we have to maintain multiple docker images per tool, for different solc versions. Moreover, smartbugs lags behind regarding solc versions. Adding a new solc version means to modify the code of smartbugs as well as to increase the number of docker images that have to be maintained.

What about writing a wrapper script `solc` that determines the Solidity version from the source code and then calls the appropriate solc version? This means that we have only one image per tool, which contains the script and one compiler per major version.

**Comment by jff** [comment]
I am currently implementing a solution for this. So far:

- I created a base docker image that uses [solc-select](https://github.com/crytic/solc-select) to install all available solc versions
- I updated smartbugs to set the version specified in the pragma as an environment variable (which solc-select / solc knows how to read)
- Preliminary tests with a new image for maian seem to work fine (i.e. by changing the pragma version, a different compiler is used)

Still to do:

- [ ] Review base image: try to reduce size as much as possible and ensure that it is based on forks that we control
- [ ] Cleanup code (e.g. we don't need to specify different images, so we simplify configuration files
- [ ] Create new images for tools that use the new base image
